src/script/main/app.ts

```diff
@@ -612,13 +613,34 @@ class App {
       }
     }
 
-    await container.resolve(StorageService).init(this.core.storage);
+    await this._initiateStorage();
+
     this.repository.client.init(userEntity);
     await this.repository.properties.init(userEntity);
 
     return userEntity;
   }
 
+  /**
+   * Initiate storage service.
+   * Trigger db migration script if relevant.
+   */
+  private async _initiateStorage() {
+    await container.resolve(StorageService).init(this.core.storage);
+
+    //storage was initialised, possible version upgrade, we can try running migration
+    const dbMigrationState = dbMigrationStateStore.getDBMigrationState();
+
+    if (dbMigrationState) {
+      try {
+        await this.core.service?.mls.proteusCryptoboxMigrate(dbMigrationState.storeName);
```

Comment:
I think it is the best place to trigger migration script - right after storage has been initialised - we're sure that version upgrade hook did run